torch/_dynamo/variables/optimizer.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict, List
+
+import torch
+from ..source import AttrSource, GetItemSource, GlobalWeakRefSource
+from ..utils import global_key_name
+
+from .base import MutableLocal, VariableTracker
+from .constant import ConstantVariable
+from .dicts import ConstDictVariable
+from .lists import ListVariable
+from .misc import GetAttrVariable
+from .user_defined import UserDefinedObjectVariable
+
+
+class ArgMappingException(Exception):
+    pass
+
+
+class OptimizerVariable(UserDefinedObjectVariable):
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        """This is an optimization to avoid tracing the very slow intialization of the optimizer"""
+        if name == "_init_group":
+            try:
+                py_args, py_kwargs = self.get_python_args(*args, **kwargs)
+                self.value._init_group(*py_args, **py_kwargs)
+                self.install_guards(tx)
+                self.update_list_args(tx, args, kwargs, py_args, py_kwargs)
+                return ConstantVariable(None)
+            except ArgMappingException:
+                # trace normally if we can't map args
+                pass
+
+        return super().call_method(tx, name, args, kwargs)
+
+    def var_getattr(self, tx, name):
+        if name == "_init_group":
+            return GetAttrVariable(self, name)
+
+        return super().var_getattr(tx, name)
+
+    def get_python_args(self, *args, **kwargs):
+        """Get python values equivalent to the variable tracker args"""
+
+        def map_arg(arg):
+            if isinstance(arg, ConstantVariable):
+                return arg.as_python_constant()
+            elif isinstance(arg, ListVariable) and not arg.items:
+                return []
+            elif isinstance(arg, ConstDictVariable) and isinstance(
+                arg.source, GetItemSource
```

Comment:
I think we need to check that arg.source is actually pointing to the param_groups here.